testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt

```diff
@@ -99,17 +100,32 @@ class SettingsMenuActivity : BaseActivity() {
         editAppId.addTextChangedListener(settingsTextWatcher)
         editSubscriptionKey.addTextChangedListener(settingsTextWatcher)
 
-        enableSaveView()
+        validateInputIDs()
     }
 
     private fun createSettingsInfo(): String {
         return "Build " + getString(R.string.miniapp_sdk_version) + " - " +
                 getString(R.string.build_version)
     }
 
-    private fun enableSaveView() {
-        saveViewEnabled =
-            !(editAppId.text.toString().isInvalidUuid() || editSubscriptionKey.text.isNullOrEmpty())
+    internal fun validateInputIDs() {
+        val isAppIdInvalid = editAppId.text.toString().isInvalidUuid()
```

Comment:
The validation logic can be separated from fetching the values and setting error values.

Reply:
Yes, I agree to have SRP principle per operation e.g. fetching/setting values. In this PR, I focused to write the validation operation in a single function. We can separate the operations precisely while adding more spec in future.